spd_conn.cc

```diff
@@ -255,6 +274,17 @@ void spider_free_conn_from_trx(
             pthread_mutex_unlock(&spider_conn_mutex);
             spider_free_conn(conn);
           } else {
+            long mutex_num=0;
+            if(ip_port_conn = (SPIDER_IP_PORT_CONN*) my_hash_search_using_hash_value(&spider_ipport_conns, conn->conn_key_hash_value, (uchar*) conn->conn_key, conn->conn_key_length))
+            {/* exists */
+              if(spider_param_max_connections())
```

Comment:
If spider_max_connections is changed to 0 dynamically, all waiting threads get timeout. I need to think about this behavior.